### PR TITLE
Updating the json.net package

### PR DESCRIPTION
**Pull request by HEskandari**

### CODE REVIEW

File: src/MsSqlAcceptanceTests/MsSqlAcceptanceTests.csproj [hunk 1]
```diff
@@ -21,7 +21,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="6.4.4" />
     <PackageReference Include="NServiceBus.SqlServer" Version="3.*" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/MsSqlAcceptanceTests/MsSqlAcceptanceTests.csproj]: Don't add version ranges on projects that don't produce a package.

File: src/MySqlAcceptanceTests/MySqlAcceptanceTests.csproj [hunk 2]
```diff
@@ -20,7 +20,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="6.4.4" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/MySqlAcceptanceTests/MySqlAcceptanceTests.csproj]: Don't add version ranges on projects that don't produce a package.

File: src/OracleAcceptanceTests/OracleAcceptanceTests.csproj [hunk 3]
```diff
@@ -20,7 +20,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="6.4.4" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/OracleAcceptanceTests/OracleAcceptanceTests.csproj]: Don't add version ranges on projects that don't produce a package.

File: src/PostgreSqlAcceptanceTests/PostgreSqlAcceptanceTests.csproj [hunk 4]
```diff
@@ -21,7 +21,7 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="NServiceBus.AcceptanceTests.Sources" Version="6.4.4" />
     <PackageReference Include="NServiceBus.SqlServer" Version="3.*" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/PostgreSqlAcceptanceTests/PostgreSqlAcceptanceTests.csproj]: Don't add version ranges on projects that don't produce a package.

File: src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj [hunk 5]
```diff
@@ -16,10 +16,10 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/ScriptBuilder.Tests/ScriptBuilder.Tests.csproj]: Don't add version ranges on projects that don't produce a package.

File: src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj [hunk 6]
```diff
@@ -14,9 +14,9 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/ScriptBuilderTask.Tests/ScriptBuilderTask.Tests.csproj]: Don't add version ranges on projects that don't produce a package.

File: src/SqlPersistence.Tests/SqlPersistence.Tests.csproj [hunk 7]
```diff
@@ -18,9 +18,9 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="NServiceBus" Version="6.*" />
     <PackageReference Include="NServiceBus.SqlServer" Version="3.*" />
-    <PackageReference Include="NUnit" Version="3.7.1" />
+    <PackageReference Include="NUnit" Version="[3.11.0, 4.0.0)" />
```
Comment (bording) [inline src/SqlPersistence.Tests/SqlPersistence.Tests.csproj]: Don't add version ranges on projects that don't produce a package.

**Comment by HEskandari** [comment]
@bording the reason to update the tests project to net461 is this:

- Test projects use ObjectApproval
- ObjectApproval v2.2 (currently references) supports net452 but uses Json.net v11
- ObjectApproval v3 (updated reference) that uses Json.net v12 does not support net452 and only supports net461.

So we either had to add Json.net v12 support to the v2.2 branch of ObjectApproval (which was refused by the project owner), or as a compromise just update the test projects to net 461 to make it work.

Given this is only a support branch and we won't have this issue on the master/develpo, I thought this is good trade-off (also checked the the maintainers).

**Comment by bording** [comment]
@HEskandari Thanks for the explanation. If it's not desirable to move to Particular.Approvals on this branch, then bumping to `net461` makes sense.

**Comment by andreasohlund** [comment]
Is this PR's still relevant? (I'm about to convert this repo to release flow)

**Comment by andreasohlund** [comment]
3.2 is no longer supported, closing this

**Pull request closed without merging**